dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AdminPrivilegesFilter.java

```diff
@@ -0,0 +1,36 @@
+package io.dockstore.webservice.resources;
+
+import java.io.IOException;
+import java.security.Principal;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+
+import io.dockstore.webservice.core.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Provider
+public class AdminPrivilegesFilter implements ContainerResponseFilter {
```

Comment:
Should this be a `ContainerRequestFilter` instead? I think the way you have it now it's logging the request when it's sending the response. Ideally you want to log the request right when it comes in. It's a stretch, but if some admin request request crashed the server before the response could be sent, nothing would be logged.

You need for the principal to have been set though, or it won't work, so you'll have to see if it's feasible.